src/plugins/credential_management/common/types.ts

```diff
@@ -9,19 +9,25 @@
  * GitHub history for details.
  */
 
-interface ICredential {
+// TODO: refactor the credential in heritance:
+// Credential -> USERNAMEANDPASSWORDCredential
+//            -> AWSIAMCredential
+export interface ICredential {
   readonly credential_name: string;
-  readonly credential_type: string;
+  readonly credential_type: CredentialType;
   readonly credential_material: IBasicAuthCredentialMaterial | IAWSIAMCredentialMaterial;
 }
 
-interface IBasicAuthCredentialMaterial {
+export type CredentialType = USERNAME_PASSWORD_TYPE | AWS_IAM_TYPE;
+// TODO: Update server side
+export type USERNAME_PASSWORD_TYPE = 'username_password_credential';
+export type AWS_IAM_TYPE = 'aws_iam_credential';
+
+export interface IBasicAuthCredentialMaterial {
   readonly user_name: string;
   readonly password: string;
 }
 
-interface IAWSIAMCredentialMaterial {
+export interface IAWSIAMCredentialMaterial {
   readonly encrypted_aws_iam_credential: string;
```

Comment:
how do we use this `encrypted_aws_iam_credential`?

Reply:
Will finalize it starting from proposal - https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1742